test/custom_verbose_configuration_tests.cpp

```diff
@@ -0,0 +1,74 @@
+#include "adobe/contract_checks.hpp"
+#include "portable_death_tests.hpp"
+#include <array>
+#include <cstdint>
+#include <cstdio>
+#include <exception>
+#include <gmock/gmock.h>
+#include <gtest/gtest.h>
+#include <stdexcept>
+
+[[noreturn]] void adobe_contract_violated_verbose(const char *condition,
+  adobe::contract_violation_kind kind,
+  const char *file,
+  std::uint32_t line,
+  const char *message)
+{
+  std::array<char, 1024> output{};
+  (void)std::snprintf(output.data(),
+    sizeof(output),
+    "%s:%d: %s (%s). %s\n",
+    file,
+    static_cast<int>(line),
+    kind == adobe::contract_violation_kind::precondition ? "Precondition violated"
+                                                         : "Invariant not upheld",
+    condition,
+    message);
+  throw std::logic_error(output.data());
+}
+
+namespace {
+
+template<class F> void expect_throw(F f, const char *match)
+{
+  try {
+    f();
+  } catch (const std::exception &e) {
+    EXPECT_THAT(e.what(), testing::ContainsRegex(match));
+    return;
+  }
+}
+
+}// namespace
+
+#define STRINGIZE(x) STRINGIZE2(x)// NOLINT(cppcoreguidelines-macro-usage)
+#define STRINGIZE2(x) #x// NOLINT(cppcoreguidelines-macro-usage)
+#define LINE_STRING STRINGIZE(__LINE__)// NOLINT(cppcoreguidelines-macro-usage)
+
+TEST(CustomVerboseConfigurationDeathTest, OneArgumentFormsAbortWithCorrectOutput)
+{
+  // clang-format off
+  expect_throw([] { ADOBE_PRECONDITION(false); }, "custom_verbose_configuration_tests\\.cpp:" LINE_STRING
+               ": Precondition violated \\(false\\)\\. \n");
+  expect_throw([] { ADOBE_INVARIANT(false); }, "custom_verbose_configuration_tests\\.cpp:" LINE_STRING
+               ": Invariant not upheld \\(false\\)\\. \n");
+  // clang-format on
+}
+
+TEST(CustomVerboseConfigurationDeathTest, TwoArgumentFormsAbortWithCorrectOutput)
+{
+  // clang-format off
+  expect_throw([] { ADOBE_PRECONDITION(false, "% Message %"); }, "custom_verbose_configuration_tests\\.cpp:" LINE_STRING
+               ": Precondition violated \\(false\\)\\. % Message %\n"); 
+  expect_throw([] { ADOBE_INVARIANT(false, "% Message %"); }, "custom_verbose_configuration_tests\\.cpp:" LINE_STRING
+               ": Invariant not upheld \\(false\\)\\. % Message %\n");
+  // clang-format on
+}
+
+TEST(CustomVerboseConfiguration, ContractNonViolationsDoNotCauseAbort)
+{
+  ADOBE_PRECONDITION(true);
+  ADOBE_INVARIANT(true);
+  ADOBE_PRECONDITION(true, "% Message %");
```

Comment:
How are you choosing these messages?  They're not consistent between the two files, which makes it seem like they might be significant.

Reply:
Which two files? This matches the VerboseConfigurationDeathTest(s).

Reply:
I saw a message that was "####" or something in one of the files, did I not?

Reply:
Bad idea to ask a question and then mark the conversation resolved, I think.

Reply:
Ah - yes, lightweight and custom lightweight tests use "####" and verbose and custom verbose use "% Message %". The distinction (I believe, you started the pattern) is that the messages aren't stored or checked in the lightweight tests.

Reply:
OK; I don't know that it makes sense to keep a distinction, notwithstanding the precedent I set.

Reply:
Fixed in the other PR (by accident - really hating multi-PR workflows).

Reply:
Changed to always use "% Message %"